cachi2/core/package_managers/pip.py

```diff
@@ -201,10 +201,16 @@ def fetch_pip_source(request: Request) -> RequestOutput:
 
 
 def _generate_properties(dependency: dict[str, str]) -> list[Property]:
+    properties = []
     if not dependency["hash_verified"]:
-        return [Property(name="cachi2:missing_hash:in_file", value=dependency["requirement_file"])]
-    else:
-        return []
+        properties.append(
+            Property(name="cachi2:missing_hash:in_file", value=dependency["requirement_file"])
```

Comment:
This comment is not specific to your changes, but just about consistency between handling properties across different package managers: Is there a reason why we don't create a PropertySet here like we do in both gomod and npm?
https://github.com/containerbuildsystem/cachi2/blob/0564776e552e286d7f2346d70ed079b83a5756fe/cachi2/core/package_managers/gomod.py#L149
https://github.com/containerbuildsystem/cachi2/blob/0564776e552e286d7f2346d70ed079b83a5756fe/cachi2/core/package_managers/npm.py#L655

Reply:
I wasn't aware that that was even a thing... I looked at 'property_semantics.py' and 'sbom.py'.

I'll change both the pip properties to use `PropertySet`.

Reply:
Fixed